README.md

```diff
@@ -13,8 +13,28 @@ Portable and scalable, it abstracts details of hardware and OS,
 enabling developers to reuse their code and knowledge on different
 targets.
 
-
-### General Information
+## TOC ##
+
+ * [General Information](#general-information)
+ * [Debug](#debug)
+ * [Libraries](#libraries)
+  * [Common](#common)
+  * [Comms](#comms)
+    * [Network](#network)
+    * [COAP](#coap)
+    * [OIC](#oic)
+  * [Flow](#flow)
+ * [Main Loops](#main-loops)
+  * [GLib](#glib-kconfig-core-library---mainloop---glib)
+  * [POSIX](#posix-kconfig-core-library---mainloop---posix)
+ * [Platforms](#platforms)
+  * [Systemd](#systemd-kconfig-core-library---target-platform---systemd)
+  * [Linux-micro](#linux-micro-kconfig-core-library---target-platform---linux-micro)
+ * [Flow Based Programming](#flow-based-programming)
+ * [Contributing](#contributing)
+
+
+## General Information
 
 **Soletta Project** uses `sol` as C namespace, so macros start with `SOL_`
```

Comment:
I'm going to split it in two sections, adding a specific about "building from source"